src/main/java/com/brightpath/learnify/domain/quiz/QuizService.java

```diff
@@ -155,6 +156,25 @@ private boolean lastIncorrectQuestionsAreTheSame(Set<QuestionEntity> oldIncorrec
         return oldIds.equals(newIncorrectIds);
     }
 
+    public Quiz updateQuizDetailsById(UUID quizId, QuizCreationDto quizCreationDto, String userId) {
+        if (findQuizEntity(quizId).isEmpty()) {
+            throw new ResourceNotFoundException(QUIZ);
+        }
+        QuizEntity quiz = entityManager.getReference(QuizEntity.class, quizId);
```

Comment:
So we are finding the quiz only to check if it exist and still operate on reference?

Reply:
missed it, right